dev/gendoc.py

```diff
@@ -0,0 +1,195 @@
+        else:
+            print("Unable to fetch URL, exiting: %s" % url, sys.stderr)
+        raise
+
+
+def list_releases_to_document(cur_version):
+    """
+    >>> list_releases_to_document("0.1.0")[0][0]
+    'Version 0.1.0'
+    """
+    tag_url = "https://api.github.com/repos/databricks/koalas/releases"
+    cur_version = "v" + cur_version
+    releases = [(
+        release['name'], release['tag_name'], release['body']) for release in get_json(tag_url)]
+    filtered = filter(
+        lambda release: LooseVersion(release[1]) <= LooseVersion(cur_version), releases)
+    return sorted(filtered, reverse=True, key=lambda release: LooseVersion(release[1]))
+
+
+def gen_release_notes(path):
+    """
+    Generate reStructuredText files for "Release Notes". It generates 'index.rst' file and
+    each rst file for each version's release note.under 'whatsnew' directory.
+    The contents are from Github release notes.
+    """
+    whatsnew_dir = "%s/whatsnew" % path
+    shutil.rmtree(whatsnew_dir, ignore_errors=True)
+    os.mkdir(whatsnew_dir)
+
+    with open("%s/index.rst" % whatsnew_dir, "a") as index_file:
+        title = "Release Notes"
+
+        index_file.write("=" * len(title))
+        index_file.write("\n")
+        index_file.write(title)
+        index_file.write("\n")
+        index_file.write("=" * len(title))
+        index_file.write("\n")
+        index_file.write("\n")
+        index_file.write(".. toctree::")
+        index_file.write("   :maxdepth: 1")
+        index_file.write("\n")
+        index_file.write("\n")
+
+        for name, tag_name, body in list_releases_to_document(ks.__version__):
+            release_doc = pypandoc.convert_text(body, "rst", format="md")
+
+            # Make PR reference link pretty.
+            # Replace ", #..." to ", `...<https://github.com/databricks/koalas/pull/...>`_"
+            release_doc = re.sub(
+                r', #(\d+)',
+                r', `#\1 <https://github.com/databricks/koalas/pull/\1>`_', release_doc)
+            # Replace "(#..." to "(`...<https://github.com/databricks/koalas/pull/...>`_"
+            release_doc = re.sub(
+                r'\(#(\d+)',
+                r'(`#\1 <https://github.com/databricks/koalas/pull/\1>`_', release_doc)
+
+            index_file.write("   " + tag_name)
+            index_file.write("\n")
+            index_file.write("\n")
+
+            with open("%s/%s.rst" % (whatsnew_dir, tag_name), "a") as release_file:
+                release_file.write("=" * len(name))
+                release_file.write("\n")
+                release_file.write(name)
+                release_file.write("\n")
+                release_file.write("=" * len(name))
+                release_file.write("\n")
+                release_file.write("\n")
+                release_file.write(release_doc)
+                release_file.write("\n")
+                release_file.write("\n")
+
+
+def download_pandoc_if_needed(path):
```

Comment:
This whole function was basically just ported from pypandoc with some minor changes, in order to:
1. check if already downloaded
2. to patch `https://github.com/bebraw/pypandoc/pull/154`